lib/html/proofer/checks/opengraph.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+
+class OpengraphCheckable < ::HTML::Proofer::Checkable
+
+  def src
+    @content
+  end
+
+  def missing_src?
+    !src
+  end
+
+  def url
+    src
+  end
+end
+
+class OpengraphCheck < ::HTML::Proofer::CheckRunner
+  def run
+    @html.css('meta[property="og:url"], meta[property="og:image"]').each do |m|
+      opengraph = OpengraphCheckable.new m, self
+
+      next if opengraph.ignore?
+
+      # does the opengraph exist?
+      if opengraph.missing_src?
+        add_issue('open graph is empty and has no content attribute', m.line)
+      elsif opengraph.remote?
+        add_to_external_urls opengraph.src, m.line
+      else
+        add_issue("internal open graph #{opengraph.src} does not exist", m.line) unless opengraph.exists?
```

Comment:
I can’t figure out how to test this line. The specs below doesn’t work.

```html
<meta property="og:image" content="doesnotexist.png" />
```

```ruby
 it 'fails for missing internal images' do
    image_internal_invalid = "#{FIXTURES_DIR}/opengraph/image-internal-invalid.html"
    proofer = run_proofer(internalImageFilepath)
    expect(proofer.failed_tests.first).to match(/doesnotexist.png does not exist/)
  end
#
Failure/Error: expect(proofer.failed_tests.first).to eq ""
     
       expected: ""
            got: nil
     
       (compared using ==)

```

@gjtorikian I need help.